### PR TITLE
Fix service exposing and private secret name

### PR DESCRIPTION
**Pull request by ausias-armesto**
- Fix service exposing of hoprd node for P2P
- Fix using a different private key name, was making the operator to restart on deleting a node

### CODE REVIEW

File: src/hoprd_secret.rs [hunk 1]
```diff
@@ -98,10 +98,10 @@ impl SecretManager {
...
 
-    /// Gets the wallet secret linked to the hoprd-operator
+    /// Gets the Kubernetes secret used by the hoprd-operator to register and fund nodes 
     pub async fn get_wallet_secret(&self) -> Result<Secret, Error> {
         let api: Api<Secret> = Api::namespaced(self.client.clone(), &self.operator_config.instance.namespace);
-        if let Some(wallet_secret) = api.get_opt(&self.operator_config.instance.name).await? {
+        if let Some(wallet_secret) = api.get_opt(&self.operator_config.instance.secret_name).await? {
             Ok(wallet_secret)
         } else {
             Err(Error::SecretStatusError("[ERROR] Could not get wallet secret".to_owned()))
```
Comment (Teebor-Choka) [inline src/hoprd_secret.rs:107]: This code should not be called. The `?` should short curcuit the error. Remove the `?` if this logic should be applied.

**Review by Teebor-Choka — APPROVED** [review]

File: src/hoprd_service.rs [hunk 2]
```diff
@@ -44,6 +38,37 @@ pub async fn create_service(client: Client, name: &str, namespace: &str, owner_r
...
+
+    service_ports.push(ServicePort {
+                name: Some("p2p-udp".to_owned()),
+                port: 9001,
+                protocol: Some("UDP".to_owned()),
+                target_port: Some(IntOrString::String("p2p-udp".to_owned())),
+                ..ServicePort::default()
+            });
+
+    return service_ports;
```
Comment (Teebor-Choka) [inline src/hoprd_service.rs:68]: Why not:
```suggestion
    vec![
        ServicePort {
                name: Some("api".to_owned()),
                port: 3001,
                protocol: Some("TCP".to_owned()),
                target_port: Some(IntOrString::String("api".to_owned())),
                ..ServicePort::default()
            },
            ...
        ServicePort {
                name: Some("p2p-tcp".to_owned()),
                port: 9001,
                protocol: Some("TCP".to_owned()),
                target_port: Some(IntOrString::String("p2p-tcp".to_owned())),
                ..ServicePort::default()
            }
        ]
```

File: src/hoprd_secret.rs (hunk 1 above)
Comment (ausias-armesto) [reply src/hoprd_secret.rs:107]: I think that you already mentioned this issue before. The template is taken from the sample code of the get_opt function:

    /// [Get](`Api::get`) a named resource if it exists, returns [`None`] if it doesn't exist
    ///
    /// ```no_run
    /// # use kube::Api;
    /// use k8s_openapi::api::core::v1::Pod;
    ///
    /// # async fn wrapper() -> Result<(), Box<dyn std::error::Error>> {
    /// # let client: kube::Client = todo!();
    /// let pods: Api<Pod> = Api::namespaced(client, "apps");
    /// if let Some(pod) = pods.get_opt("blog").await? {
    ///     // Pod was found
    /// } else {
    ///     // Pod was not found
    /// }
    /// # Ok(())
    /// # }
    /// ```
    
    I'm just using the same pattern

File: src/hoprd_service.rs [hunk 3]
```diff
@@ -44,6 +38,37 @@ pub async fn create_service(client: Client, name: &str, namespace: &str, owner_r
...
+                name: Some("api".to_owned()),
+                port: 3001,
+                protocol: Some("TCP".to_owned()),
+                target_port: Some(IntOrString::String("api".to_owned())),
+                ..ServicePort::default()
+            });
+
+    service_ports.push(ServicePort {
+                name: Some("p2p-tcp".to_owned()),
+                port: 9001,
```
[the remark this answers is not in the document]
Comment (ausias-armesto) [reply src/hoprd_service.rs]: You're right. Changed

**Pull request merged**